src/zarr/store/remote.py

```diff
@@ -1,7 +1,10 @@
 from __future__ import annotations
 
+from collections.abc import AsyncGenerator
 from typing import TYPE_CHECKING, Any
 
+import fsspec
```

Comment:
can we do this import in `RemoteStore.__init__`? Would help us keep this an optional dep here.

Reply:
Since fsspec is very small and itself has no further dependencies, making it required will reduce friction for users

Reply:
I suppose this is fine -- at least for now.